### PR TITLE
Add SelectionList component to DRY up

### PR DESCRIPTION
**Pull request by jacogr**
- Add `ui/SelectionList`
- Convert `~/modals/DappPermissions` & `~/modals/AddDapp` removing selection duplication
- Convert `~/modals/VaultAccounts` (addressing code TODO re duplication)
- Convert `~/views/ParityBar` to use selection component
- Functionality exactly the same, DRY component underneath to take care of selection & default rendering

parity 2017-02-22 17-16-33

parity 2017-02-22 17-16-18

parity 2017-02-22 17-37-30

parity 2017-02-23 10-22-48

### CODE REVIEW

**Comment by derhuerst** [comment]
I think left-aligned boxes would give it a cleaner look.

**Comment by jacogr** [comment]
Actually it is the standard layout for the `~/ui/SectionList` component, as used pre-PR-change for the affected selectors. In short, have not mucked about with that. 

Happy to have that logged and play around, obviously impacts everywhere that is used - Home, Vaults, these selectors.

File: js/src/modals/DappPermissions/dappPermissions.js [hunk 1]
```diff
@@ -61,60 +61,36 @@ class DappPermissions extends Component {
...
-
-    const onMakeDefault = () => {
-      permissionStore.setDefaultAccount(account.address);
-    };
-
-    const onSelect = () => {
-      permissionStore.selectAccount(account.address);
-    };
+  onMakeDefault = (account) => {
+    console.log('onMakeDefault', account);
```
Comment (ngotchac) [inline js/src/modals/DappPermissions/dappPermissions.js]: Might want to get rid of this one.

File: js/src/modals/DappPermissions/dappPermissions.js [hunk 2]
```diff
@@ -61,60 +61,36 @@ class DappPermissions extends Component {
...
+    this.props.permissionStore.setDefaultAccount(account.address);
+  }
 
-    let className;
+  onSelect = (account) => {
+    console.log('onSelect', account);
```
Comment (ngotchac) [inline js/src/modals/DappPermissions/dappPermissions.js]: Same as above.

File: js/src/ui/SelectionList/selectionList.css [hunk 3]
```diff
@@ -33,16 +42,16 @@
...
+.selected {
+  box-shadow: inset 0 0 0 2px rgb(255, 255, 255);
+  filter: none;
+}
 
-  &:focus {
-    outline: none;
-  }
+.unselected {
+  filter: grayscale(100%);
```
Comment (ngotchac) [inline js/src/ui/SelectionList/selectionList.css]: :+1:

**Comment by jacogr** [comment]
Will look into the scrollbar issue.

**Comment by ngotchac** [comment]
One last comment:

> I liked the difference of background between a selected only and a default account. It made it a bit more obvious than just the filled star. I really like the new greyscale/colours layout though.

**Comment by jacogr** [comment]
See it as annoying, but don't have any fixes for it in mind. So feel free to log for enhancement however.

**Comment by ngotchac** [comment]
All right, will do.

**Comment by jacogr** [comment]
Logged https://github.com/ethcore/parity/issues/4666

**Pull request merged**